bigquery_etl/dependency.py

```diff
@@ -101,30 +130,31 @@ def dependency():
 )
 def show(paths: Tuple[str, ...], without_views: bool):
     """Show table references in sql files."""
-    distinct_paths = {
-        path
-        for parent in map(Path, paths or ["sql"])
-        for path in (parent.glob("**/*.sql") if parent.is_dir() else [parent])
-        if not path.name.endswith(".template.sql")  # skip templates
-    }
-    fail = False
-    for path in sorted(distinct_paths):
-        try:
-            if without_views:
-                table_references = list(extract_table_references_without_views(path))
-            else:
-                table_references = extract_table_references(path.read_text())
-        except CalledProcessError as e:
-            raise click.ClickException(f"failed to import jnius: {e}")
-        except ImportError as e:
-            raise click.ClickException(*e.args)
-        except ValueError as e:
-            fail = True
-            print(f"Failed to parse {path}: {e}", file=sys.stderr)
+    for path, table_references in _get_references(paths, without_views):
         if table_references:
             for table in table_references:
                 print(f"{path}: {table}")
         else:
             print(f"{path} contains no table references", file=sys.stderr)
-    if fail:
-        raise click.ClickException("Some paths could not be analyzed")
+
+
+@dependency.command(
+    help="Record table references in yaml. Requires Java.",
+)
+@click.argument(
+    "paths",
+    nargs=-1,
+    type=click.Path(file_okay=True),
+)
+def record(paths: Tuple[str, ...]):
+    """Record table references in yaml."""
+    for parent, group in groupby(_get_references(paths), lambda e: e[0].parent):
```

Comment:
How quickly does this run? Does it need to be parallelized?

Reply:
it takes 5 seconds in CI, so i don't think it needs to be parallelized